app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1479,13 +1480,27 @@ class BrowserTabViewModel @Inject constructor(
         viewModelScope.launch { updateBookmarkAndFavoriteState(url) }
     }
 
+    fun stripBasicAuthFromUrl(url: String): String {
+        val uri = URI(url)
+        val userInfo = uri.userInfo
+
+        if (userInfo != null) {
+            val queryStr = uri.rawQuery?.let { "?$it" } ?: ""
+            val uriFragment = uri.fragment?.let { "#$it" } ?: ""
+            val portStr = if (uri.port != -1) ":${uri.port}" else ""
+            return "${uri.scheme}://${uri.host}$portStr${uri.path}$queryStr$uriFragment"
+        }
+
+        return url
+    }
```

Comment:
Casting the url to a `URI` can throw an exception. 
We also generally use the `toUri()` extension function instead of `URI()`.

I would replace this function with:
```
    fun stripBasicAuthFromUrl(url: String): String {
        try {
            val uri = url.toUri()
            val userInfo = uri.userInfo

            if (userInfo != null) {
                val queryStr = uri.rawQuery?.let { "?$it" } ?: ""
                val uriFragment = uri.fragment?.let { "#$it" } ?: ""
                val portStr = if (uri.port != -1) ":${uri.port}" else ""
                return "${uri.scheme}://${uri.host}$portStr${uri.path}$queryStr$uriFragment"
            }
        } catch(e: Exception) {
            return url
        }
        return url
    }
```

You should also add a test for the case where it throws an exception.